tests/cmdline/commands/test_group.py

```diff
@@ -134,24 +134,146 @@ def test_delete(self, run_cli_command):
         with pytest.raises(exceptions.NotExistent):
             orm.load_group(label='group_test_delete_02')
 
-        # check nodes still exist
         for pk in node_pks:
             orm.load_node(pk)
 
-        # delete the group and the nodes it contains
+        # 4) Delete the group and the nodes it contains
         group = orm.load_group(label='group_test_delete_03')
         group.add_nodes([node_01, node_02])
         result = run_cli_command(
             cmd_group.group_delete, ['--force', '--delete-nodes', 'group_test_delete_03'], use_subprocess=True
         )
 
-        # check group and nodes no longer exist
         with pytest.raises(exceptions.NotExistent):
             orm.load_group(label='group_test_delete_03')
         for pk in node_pks:
             with pytest.raises(exceptions.NotExistent):
                 orm.load_node(pk)
 
+        # 5) Should delete an empty group even if --delete-nodes option is passed
+        group = orm.Group(label='group_test_delete_04').store()
+        result = run_cli_command(
+            cmd_group.group_delete, ['--force', '--delete-nodes', 'group_test_delete_04'], use_subprocess=True
+        )
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_04')
+
+        # 6) Should raise if a group does not exist
+        result = run_cli_command(
+            cmd_group.group_delete, ['--force', 'non_existent_group'], raises=True, use_subprocess=True
+        )
+
+        # 7) Should delete multiple groups
+        orm.Group(label='group_test_delete_05').store()
+        orm.Group(label='group_test_delete_06').store()
+        result = run_cli_command(
+            cmd_group.group_delete, ['--force', 'group_test_delete_05', 'group_test_delete_06'], use_subprocess=True
+        )
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_05')
+        with pytest.raises(exceptions.NotExistent):
+            orm.load_group(label='group_test_delete_06')
+
+        # 8) Should raise if both groups and query options are passed
+        orm.Group(label='group_test_delete_07').store()
+        result = run_cli_command(
+            cmd_group.group_delete,
+            ['--force', 'group_test_delete_01', '--all-users', 'label==group_test_delete_07'],
```

Comment:
What is this `label==group_test_delete_07`? I don't think that is a valid argument. I think the command is failing as you expect, but for a different reason. That is why it is good to always have an assert that checks the error message for a command that should fail.

Reply:
fixed